test/functional/feature_config_args.py

```diff
@@ -213,11 +213,30 @@ def test_seed_peers(self):
         ]):
             self.nodes[0].setmocktime(start + 65)
 
+    def test_addnode_and_connect_i2p(self):
+        self.log.info('Test that ports are removed from -addnode= and -connect= I2P addresses')
+
+        i2p_addr = "ukeu3k5oycgaauneqgtnvselmt4yemvoilkln7jpvamvfx7dnkdq.b32.i2p"
+        i2p_addr_with_port = i2p_addr + ":3333"
+
+        self.stop_node(0)
+        with self.nodes[0].assert_debug_log(expected_msgs=[
+                f"trying connection {i2p_addr} lastseen=",
+        ]):
+            self.start_node(0, extra_args=[f"-addnode={i2p_addr_with_port}"])
+
+        self.stop_node(0)
+        with self.nodes[0].assert_debug_log(expected_msgs=[
+                f"trying connection {i2p_addr} lastseen=",
+        ]):
+            self.start_node(0, extra_args=[f"-connect={i2p_addr_with_port}"])
```

Comment:
Taketh, thanks!

One minor downside of the new variant is that if we get exception on line 224, then it is not clear whether the problem is in `-addnode` or `-connect`.

Reply:
Good point, `loglevel=debug` (logged by the CI in case of failure) would reliably indicate which one if we use an array/dict rather than a set:

```diff
-        for option in {"addnode", "connect"}:
+        for option in ["addnode", "connect"]:
```
or by adding, for instance
```diff
         for option in {"addnode", "connect"}:
+            self.log.debug(f"-{option}")
             with self.nodes[0].assert_debug_log(expected_msgs=[msg]):
```
then the log would show
```
2021-04-01T17:42:16.846000Z TestFramework (INFO): Test ports are removed from I2P addresses passed to -addnode/-connect
2021-04-01T17:42:16.846000Z TestFramework (DEBUG): -connect
2021-04-01T17:42:16.846000Z TestFramework.node0 (DEBUG): Stopping node
2021-04-01T17:42:16.999000Z TestFramework.node0 (DEBUG): Node stopped
2021-04-01T17:42:17.002000Z TestFramework.node0 (DEBUG): bitcoind started, waiting for RPC to come up
2021-04-01T17:42:17.512000Z TestFramework.node0 (DEBUG): RPC successfully started
2021-04-01T17:42:17.514000Z TestFramework (DEBUG): -addnode
2021-04-01T17:42:17.514000Z TestFramework.node0 (DEBUG): Stopping node
2021-04-01T17:42:17.618000Z TestFramework.node0 (DEBUG): Node stopped
2021-04-01T17:42:17.628000Z TestFramework.node0 (DEBUG): bitcoind started, waiting for RPC to come up
2021-04-01T17:42:18.140000Z TestFramework.node0 (DEBUG): RPC successfully started
```

Reply:
> ...would reliably indicate which one if we use an array/dict rather than a set...

Hmm, I tried this, but did not see any difference in the output, what am I missing? (I sneaked a deliberate change of the expected message if the option is e.g. `addnode` in order to trigger a failure).

Reply:
I wasn't very clear. I meant that it would have a fixed order so you would know which failure it is. Dict + debug logging probably best.